tests/agents/gherkin-specs/api_key.feature

```diff
@@ -1,18 +1,24 @@
-Feature: API Key.
+Feature: APM server authentication with API key and secret token
 
-  Scenario: A configured API key is sent in the Authorization header.
-    Given an agent
-    When an api key is set to 'RTNxMjlXNEJt' in the config
-    Then the Authorization header is 'ApiKey RTNxMjlXNEJt'
+  Scenario: A configured API key is sent in the Authorization header
+    Given an agent configured with
+      | setting    | value        |
+      | api key    | RTNxMjlXNEJt |
```

Comment:
I think it makes sense to just use the config keys as defined for central config (lower snake case without `elastic_apm_` prefix). Agents can then just implement the "Given an agent configured with" phrase generically, without having to write adapters for each individual option.
```suggestion
      | api_key    | RTNxMjlXNEJt |
```